### PR TITLE
Implement card 'Captain Greenskin' (NEW1_024)

### PR DESCRIPTION
**Pull request by CreatorSeraph**
This revision includes:
* Implement 1 card in the EXPERT1 card set
    * Captain Greenskin (NEW1_024)
* Create function `DurabilityN()`

### CODE REVIEW

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp [hunk 1]
```diff
@@ -3271,6 +3271,16 @@ void Expert1CardsGen::AddNeutral(std::map<std::string, Power>& cards)
     power.GetTrigger()->tasks = { new DestroyTask(EntityType::ALL_MINIONS) };
     cards.emplace("NEW1_021", power);
 
+	// --------------------------------------- MINION - NEUTRAL
```
Comment (utilForever) [inline Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: You have to rearrange whitespaces.

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp [hunk 2]
```diff
@@ -3271,6 +3271,16 @@ void Expert1CardsGen::AddNeutral(std::map<std::string, Power>& cards)
...
+    // [NEW1_024] Captain Greenskin - COST:5 [ATK:5/HP:4]
+    // - Race: Pirate, Faction: Neutral, Set: Expert1, Rarity: legendary
+    // --------------------------------------------------------
+    // "Text : "<b>Battlecry:</b> Give your weapon +1/+1.",
```
Comment (utilForever) [inline Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: You have to remove double quotes("").

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp [hunk 3]
```diff
@@ -3271,6 +3271,16 @@ void Expert1CardsGen::AddNeutral(std::map<std::string, Power>& cards)
     power.GetTrigger()->tasks = { new DestroyTask(EntityType::ALL_MINIONS) };
     cards.emplace("NEW1_021", power);
 
+	// --------------------------------------- MINION - NEUTRAL
+    // [NEW1_024] Captain Greenskin - COST:5 [ATK:5/HP:4]
+    // - Race: Pirate, Faction: Neutral, Set: Expert1, Rarity: legendary
```
Comment (utilForever) [inline Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: You have to remove `Faction: Neutral, `

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp [hunk 4]
```diff
@@ -3689,6 +3699,18 @@ void Expert1CardsGen::AddNeutralNonCollect(std::map<std::string, Power>& cards)
...
+    // ---------------------------------- ENCHANTMENT - NEUTRAL
+    // [NEW1_024o] Greenskin's Command (*) - COST:0
+    // - Set: Expert1
+    // --------------------------------------------------------
+    // Text: +1/+1.
+    // --------------------------------------------------------
+    power.ClearData();
+    std::vector<Effect*> enchantEffects = { Effects::AttackN(1),
+                                            Effects::DurabilityN(1) };
+    power.AddEnchant(new Enchant(enchantEffects));
```
Comment (utilForever) [inline Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: I think that this statements can be combined.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp [hunk 5]
```diff
@@ -8402,6 +8402,50 @@ TEST(NeutralExpert1Test, NEW1_021_Doomsayer)
     EXPECT_EQ(opField.GetCount(), 0);
 }
 
+// --------------------------------------- MINION - NEUTRAL
+// [NEW1_024] Captain Greenskin - COST:5 [ATK:5/HP:4]
+// - Race: Pirate, Faction: Neutral, Set: Expert1, Rarity: legendary
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: Same as above.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp [hunk 6]
```diff
@@ -8402,6 +8402,50 @@ TEST(NeutralExpert1Test, NEW1_021_Doomsayer)
...
+// --------------------------------------------------------
+// "Text : "<b>Battlecry:</b> Give your weapon +1/+1.",
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: Same as above.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp [hunk 7]
```diff
@@ -8402,6 +8402,50 @@ TEST(NeutralExpert1Test, NEW1_021_Doomsayer)
...
+    game.ProcessUntil(Step::MAIN_START);
+
+    Player& curPlayer = game.GetCurrentPlayer();
+    Player& opPlayer = game.GetOpponentPlayer();
+    curPlayer.SetTotalMana(10);
+    curPlayer.SetUsedMana(0);
+    opPlayer.SetTotalMana(10);
+    opPlayer.SetUsedMana(0);
+
+	const auto card1 = Generic::DrawCard(
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: You have to rearrange whitespaces.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp [hunk 8]
```diff
@@ -8402,6 +8402,50 @@ TEST(NeutralExpert1Test, NEW1_021_Doomsayer)
...
+        curPlayer, Cards::GetInstance().FindCardByName("Captain Greenskin"));
+
+    game.Process(curPlayer, HeroPowerTask());
+    EXPECT_EQ(curPlayer.GetHero()->weapon->GetAttack(), 1);
+    EXPECT_EQ(curPlayer.GetHero()->weapon->GetDurability(), 2);
+
+    game.Process(curPlayer, PlayCardTask::Spell(card1));
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: `Caption Greenskin` is not spell, but minion.

File: Includes/Rosetta/Enchants/Effects.hpp [hunk 9]
```diff
@@ -30,6 +30,12 @@ class Effects
         return new Effect(GameTag::HEALTH, EffectOperator::ADD, n);
     }
 
+	//! Creates effect that increases health by \p n.
```
Comment (utilForever) [inline Includes/Rosetta/Enchants/Effects.hpp]: You have to correct comments (not `health` but `durability`) and rearrange whitespaces.

Comment (utilForever) [reply Includes/Rosetta/Enchants/Effects.hpp]: Also, you have to add comment for parameter and return type.

Comment (CreatorSeraph) [reply Includes/Rosetta/Enchants/Effects.hpp]: The other functions in Effects.hpp don't have a comment on return type or paramter.

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp (hunk 1 above)
Comment (CreatorSeraph) [reply Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: fixed.

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp (hunk 2 above)
Comment (CreatorSeraph) [reply Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: fixed.

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp (hunk 3 above)
Comment (CreatorSeraph) [reply Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: fixed.

File: Sources/Rosetta/CardSets/Expert1CardsGen.cpp (hunk 4 above)
Comment (CreatorSeraph) [reply Sources/Rosetta/CardSets/Expert1CardsGen.cpp]: fixed.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp (hunk 5 above)
Comment (CreatorSeraph) [reply Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: fixed.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp (hunk 6 above)
Comment (CreatorSeraph) [reply Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: fixed.

File: Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp (hunk 8 above)
Comment (CreatorSeraph) [reply Tests/UnitTests/CardSets/Expert1CardsGenTests.cpp]: fixed.

File: Includes/Rosetta/Enchants/Effects.hpp (hunk 9 above)
Comment (utilForever) [reply Includes/Rosetta/Enchants/Effects.hpp]: Hmm... Other functions have to add comment for parameter and return type too. Can you add it?

Comment (CreatorSeraph) [reply Includes/Rosetta/Enchants/Effects.hpp]: Ok. I'll try it.

Comment (utilForever) [reply Includes/Rosetta/Enchants/Effects.hpp]: Thanks. 😄

**Pull request merged**